api/src/flows.ts

```diff
@@ -0,0 +1,362 @@
+import * as sharedExceptions from '@directus/shared/exceptions';
+import {
+	ActionHandler,
+	FilterHandler,
+	Flow,
+	InitHandler,
+	Operation,
+	OperationHandler,
+	SchemaOverview,
+	Accountability,
+	Action,
+} from '@directus/shared/types';
+import { get } from 'micromustache';
+import { schedule, validate } from 'node-cron';
+import getDatabase from './database';
+import emitter from './emitter';
+import env from './env';
+import * as exceptions from './exceptions';
+import logger from './logger';
+import * as services from './services';
+import { FlowsService } from './services';
+import { EventHandler } from './types';
+import { constructFlowTree } from './utils/construct-flow-tree';
+import { getSchema } from './utils/get-schema';
+import { renderMustache } from './utils/render-mustache';
+import { ActivityService } from './services/activity';
+import { RevisionsService } from './services/revisions';
+import { Knex } from 'knex';
+
+let flowManager: FlowManager | undefined;
+
+export function getFlowManager(): FlowManager {
+	if (flowManager) {
+		return flowManager;
+	}
+
+	flowManager = new FlowManager();
+
+	return flowManager;
+}
+
+type TriggerHandler = {
+	id: string;
+	events: EventHandler[];
+};
+
+const TRIGGER_KEY = '$trigger';
+const ACCOUNTABILITY_KEY = '$accountability';
+const LAST_KEY = '$last';
+
+class FlowManager {
+	private operations: Record<string, OperationHandler> = {};
+
+	private triggerHandlers: TriggerHandler[] = [];
+	private operationFlowHandlers: Record<string, any> = {};
+	private webhookFlowHandlers: Record<string, any> = {};
+
+	private flowOperations: Record<string, any> = {};
+
+	public async initialize(): Promise<void> {
+		const flowsService = new FlowsService({ knex: getDatabase(), schema: await getSchema() });
+
+		const flows = await flowsService.readByQuery({
+			filter: { status: { _eq: 'active' } },
+			fields: ['*', 'operations.*'],
+		});
+
+		for (const flow of flows) {
+			for (const operationRaw of flow.operations) {
+				const operation = {
+					...operationRaw,
+					resolve: null,
+					reject: null,
+				};
+
+				const handler = (data: Record<string, unknown>) => this.executeOperation(operation, data);
+
+				this.flowOperations[operation.id] = handler;
+			}
+		}
+
+		const flowTrees = flows.map((flow) => constructFlowTree(flow));
+
+		for (const flow of flowTrees) {
+			if (flow.trigger === 'hook') {
```

Comment:
Not sure if `hook` makes that much sense here as we are not really triggering on a hook but on an event. Also, this might be confusing because `schedule` is part of hook extensions but here it is a dedicated trigger.

Reply:
Good point! I'm down to rename it to `event` 👍🏻